datastore/src/json_file.rs

```diff
@@ -54,14 +54,17 @@ where
 {
     /// Opens or creates the datastore. If the path refers to an existing path, then this function
     /// will attempt to load an existing set of values from it (which can result in an error).
-    /// Otherwise if the path doesn't exist, a new empty datastore will be created.
+    /// Otherwise if the path doesn't exist, the parent directory will be created and a new empty
+    /// datastore will be returned.
     pub fn new<P>(path: P) -> Result<JsonFileDatastore<T>, IoError>
     where
         P: Into<PathBuf>,
     {
         let path = path.into();
 
         if !path.exists() {
+            fs::create_dir_all(path.parent()
+                .expect("can only fail if root, which means that path.exists() would be true"))?;
```

Comment:
This could also fail due to some permission problem or other I/O related issue. Why not simply returning the error?

Reply:
This is the about the call to `parent()`. The call to `create_dir_all()` gets its error returned.

Reply:
Ah, yes. Sorry, internal parser error.